src/saga.js

```diff
@@ -0,0 +1,65 @@
+/* @flow */
+
+import { put, select, call, take, cancelled, fork } from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { NetInfo } from 'react-native';
+import checkInternetAccess from './checkInternetAccess';
+import { connectionChange } from './actionCreators';
+import type { NetworkState } from './types';
+
+export function* forceTryConnectivity(): Generator<*, *, *> {
+  yield call(handleConnectivityChange, true);
+}
+
+export function* networkEventsListenerSaga(
+  timeout?: number,
+  pingServerUrl?: string,
+): Generator<*, *, *> {
+  const isConnectedChannel = yield call(createNetInfoIsConnectedChannel);
+  try {
+    while (true) {
+      const isConnected = yield take(isConnectedChannel);
+      yield fork(handleConnectivityChange, isConnected, timeout, pingServerUrl);
+    }
+  } finally {
+    if (yield cancelled()) {
+      if (isConnectedChannel) {
```

Comment:
Do we really need this check?

Reply:
I'm not sure why I put this check there. Maybe I had an undefined channel once, but I can't remember. Anyway, I'll remove this check and make some tests.